File: examples/order_processor.rb
require 'pg'
require 'net/smtp'
require 'mail'

class OrderProcessor
  def process_order(order)
    raise "Order is empty" if order[:items].empty?
    
    total = order[:items].sum { |item| item[:price] * item[:quantity] }
    
    if order[:customer_type] == "Premium"
      total *= 0.9  # 10% discount
    end
    
    # Save to database
    connection = PG.connect(
      host: 'localhost',
      dbname: 'mydb',
      user: 'postgres',
      password: 'password'
    )
    connection.exec_params(
      "INSERT INTO Orders (CustomerId, Total, CustomerType) VALUES ($1, $2, $3)",
      [order[:customer_id], total, order[:customer_type]]
    )
    connection.close
    
    # Send email
    mail = Mail.new do
      from    'your-email@gmail.com'
      to      'customer@email.com'
      subject 'Order confirmed'
      body    'Your order has been processed'
    end
    
    mail.delivery_method :smtp, {
      address: 'smtp.gmail.com',
      port: 587,
      domain: 'gmail.com',
      user_name: 'your-email@gmail.com',
      password: 'your-password',
      authentication: 'plain',
      enable_starttls_auto: true
    }
    
    mail.deliver!
  end
end

File: examples/order_processor.py
import psycopg2
import smtplib
from email.mime.text import MIMEText

class OrderProcessor:
    def process_order(self, order: dict) -> None:
        if not order["items"]:
            raise ValueError("Order is empty")
        
        total = sum(item["price"] * item["quantity"] for item in order["items"])
        
        if order["customer_type"] == "Premium":
            total *= 0.9  # 10% discount
        
        # Save to database
        connection = psycopg2.connect(
            host="localhost",
            database="mydb",
            user="postgres",
            password="password"
        )
        # ... save order
        connection.close()
        
        # Send email
        smtp = smtplib.SMTP('smtp.gmail.com', 587)
        smtp.starttls()
        smtp.login('your-email@gmail.com', 'your-password')
        msg = MIMEText('Your order has been processed')
        msg['Subject'] = 'Order confirmed'
        msg['From'] = 'your-email@gmail.com'
        msg['To'] = 'customer@email.com'
        smtp.send_message(msg)
        smtp.quit()

